inference-engine/src/mkldnn_plugin/mkldnn_memory.cpp

```diff
@@ -125,15 +125,26 @@ void MKLDNNMemory::SetData(memory::data_type dataType, memory::format format, co
     }
 }
 
-void MKLDNNMemory::SetData(const MKLDNNMemory& memory, bool ftz) const {
```

Comment:
The second impl of SetData assumes MKLDNNMemory object creation which is not happening for all branches in the second impl. In other words this may change the overall logic and lead to unpredictable effects.
In general I agree that this part should be reworked, but I won't do it in bounds of this PR.